tree-sitter-stack-graphs/src/cli/index.rs

```diff
@@ -354,6 +334,56 @@ impl<'a> Indexer<'a> {
         Ok(())
     }
 
+    fn build_stack_graph<'b>(
+        graph: &mut StackGraph,
+        file: Handle<File>,
+        source_root: &Path,
+        source_path: &Path,
+        source: &'b str,
+        lcs: FileLanguageConfigurations<'b>,
+        cancellation_flag: &dyn CancellationFlag,
+    ) -> std::result::Result<(), BuildErrorWithSource<'b>> {
+        let relative_source_path = source_path.strip_prefix(source_root).unwrap();
+        if let Some(lc) = lcs.primary {
+            let globals = Variables::new();
+            lc.sgl
+                .build_stack_graph_into(graph, file, source, &globals, cancellation_flag)
+                .map_err(|inner| BuildErrorWithSource {
+                    inner,
+                    source_path: source_path.to_path_buf(),
+                    source_str: source,
+                    tsg_path: lc.sgl.tsg_path().to_path_buf(),
+                    tsg_str: &lc.sgl.tsg_source(),
+                })?;
+        }
+        for lc in lcs.secondary {
+            let fa = match source_path
+                .file_name()
+                .and_then(|file_name| lc.special_files.get(&file_name.to_string_lossy()))
+            {
+                Some(fa) => fa,
+                None => continue, // shouldn't really happen though
```

Comment:
Can we log “invariant violated” or something?

Reply:
First I was going to turn this into a panic. But now I've changed it so that the loader returns not only a reference to the language configuration, but also the specific analyzer for the file. This should remove any possible confusion on (a) where to get the analyzer, and (b) whether that lookup could still fail.